Add the ability to skip virtualenv setup when using an existing one

LukeZielinski opened the pull request:
This allows WPT to use a virtualenv directory that has already been setup.

We can choose between three configurations via the --venv and --skip-venv-setup flags
1) setting up a virtualenv in the default "_venv" directory
2) setting up a virtualenv in a custom directory (by setting --venv only)
3) skipping all virtualenv setup and using an existing directory (by setting --venv and --skip-venv-setup together).

LukeZielinski committed:
Add the ability to skip setting up the virtualenv if one is already
available.

LukeZielinski commented:
@jgraham: A while ago we talked about making virtualenv setup optional. This may be too hacky, but let me know what you think.

LukeZielinski commented on tools/wpt/virtualenv.py:
```diff
@@ -8,18 +8,40 @@
-        if not self.virtualenv:
-            raise ValueError("virtualenv must be installed and on the PATH")
+        self.skip_virtualenv_setup = skip_virtualenv_setup
+        if skip_virtualenv_setup:
```
My fault, I didn't realize that the activation script does the path adjustment, so this can be removed.

In my case, though, the virtualenv has already been activated by the time WPT runs, so the activation script is not there. How about we try to run the activation script, and if it fails (likely because the file doesn't exist), we ignore the error when `skip_virtualenv_setup` is True.

jgraham replied:
Eh? The file does still exist when the virtualenv has been activated. Are you actually running something that isn't really virtualenv and so behaves slightly differently?

LukeZielinski replied:
So on my end, there is a virtualenv on the machine but it's in a separate location from where WPT lives, and $PATH is already updated with the right directories. So I should actually be able to locate this file via $PATH, but I don't need to because it's already been activated. So I don't actually need any of this path updating logic, sorry for the churn.

In general it seems safe to assume that the virtualenv has been activated if we're skipping its setup. Let me know if you disagree.

jgraham commented on tools/wpt/virtualenv.py:
```diff
@@ -38,16 +60,24 @@ def pip_path(self):
         return path
 
     def activate(self):
+        if self.skip_virtualenv_setup:
```
I think I would prefer if the class didn't have to know at all about the way it's going to be called and we just call whichever lifecycle methods are required given the initial state.

LukeZielinski replied:
Done.

LukeZielinski commented on tools/wpt/wpt.py:
```diff
@@ -77,15 +80,19 @@ def import_command(prog, command, props):
-        venv.install_requirements(path)
+    venv = virtualenv.Virtualenv(path, should_skip_setup)
+    if not should_skip_setup:
+        venv.start()
```
It'd be one or the other, right? If we remove the changes in `virtualenv.py` then we need this function to skip calling `start` and `install`. If we keep the changes in `virtualenv.py` then this function can just call all these methods since they'd be no-ops.

I agree we probably don't need both, so I'll change this based on the other comment above.

LukeZielinski replied:
Done

LukeZielinski committed:
Remove skipping logic from virtualenv.py, let the caller decide which
methods to call.

Also don't bother with any activation, assume that a well-setup
virtualenv has already been activated.

LukeZielinski reviewed (commented):
Thanks James.

jgraham commented on tools/wpt/virtualenv.py:
```diff
@@ -9,11 +9,13 @@
-            raise ValueError("virtualenv must be installed and on the PATH")
+        self.skip_virtualenv_setup = skip_virtualenv_setup
+        if not skip_virtualenv_setup:
+            self.virtualenv = find_executable("virtualenv")
```
So this sort of leaves the class in a semi-initalised state. I guess it's OK for now, but it doesn't seem perfect.

LukeZielinski committed:
Fix test failure by adding skip_virtualenv_setup attribute to test venv

LukeZielinski committed:
Merge branch 'master' into skip-venv-setup
